CLI/itkSignalIntensityToConcentrationImageFilter.hxx

```diff
@@ -198,6 +131,98 @@ void SignalIntensityToConcentrationImageFilter<TInputImage, TMaskImage, TOutput>
   os << indent << "S0GradThresh: " << m_S0GradThresh << std::endl;
 }
 
-} // end namespace itk
 
+
+//==================== T1PreValueIterator internal helper class ====================
+
+template<class TInputImage, class TMaskImage, class TOutputImage>
+SignalIntensityToConcentrationImageFilter<TInputImage, TMaskImage, TOutputImage>::T1PreValueIterator::T1PreValueIterator(const InputMaskType* roiMask,
+                                                                                                                         const InputMaskType* aifMask, 
+                                                                                                                         const InputMaskType* t1Map, 
+                                                                                                                         float t1PreTissue, 
+                                                                                                                         float t1PreBlood)
+{
+  this->m_T1PreTissue = t1PreTissue;
+  this->m_T1PreBlood = t1PreBlood;
+  this->roiMaskVolumeIter = this->getNewConstMaskIterOrNull(roiMask);
+  this->aifMaskVolumeIter = this->getNewConstMaskIterOrNull(aifMask);
+  this->T1MapVolumeIter = this->getNewConstMaskIterOrNull(t1Map);
+}
+
+
+template<class TInputImage, class TMaskImage, class TOutputImage>
+SignalIntensityToConcentrationImageFilter<TInputImage, TMaskImage, TOutputImage>::T1PreValueIterator::~T1PreValueIterator()
+{
+  delete this->roiMaskVolumeIter;
+  delete this->aifMaskVolumeIter;
+  delete this->T1MapVolumeIter;
+}
+
+
+template<class TInputImage, class TMaskImage, class TOutputImage>
+float
+SignalIntensityToConcentrationImageFilter<TInputImage, TMaskImage, TOutputImage>::T1PreValueIterator::Get()
```

Comment:
Can you document the essence of this function alongside the definition of this class?

I would also consider naming this class "Mapper" instead of "Iterator", since it does more than iterating over the map voxels. "Iterator" can be misleading.